### PR TITLE
Eliminate `experimental` feature

### LINKED ISSUE #3648: Eliminate `experimental` feature
As [proposed](https://github.com/polkadot-fellows/runtimes/pull/228#discussion_r1519618615) by @bkchr, remove every mention of the `experimental` feature from the entire codebase for the sake of reducing the entropy of the Universe.

### PR DESCRIPTION
**Pull request by s0me0ne-unkn0wn**
Totally removes the `experimental` feature. Closes #3648.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5499178

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/5499179

File: substrate/frame/support/src/lib.rs [hunk 1]
```diff
@@ -177,8 +177,7 @@ pub use frame_support_procedural::storage_alias;
 
 pub use frame_support_procedural::derive_impl;
 
-/// Experimental macros for defining dynamic params that can be used in pallet configs.
-#[cfg(feature = "experimental")]
```
Comment (ggwpez) [inline substrate/frame/support/src/lib.rs:181]: FRAME is still using an experimental stage for its upcoming features, please revert.

Comment (kianenigma) [reply substrate/frame/support/src/lib.rs:181]: Yeah, we do want this in FRAME.

Comment (s0me0ne-unkn0wn) [reply substrate/frame/support/src/lib.rs]: I are not thinks! I works! I removes stuff!

Comment (s0me0ne-unkn0wn) [reply substrate/frame/support/src/lib.rs]: Reverted everything but Aura, sorry for misunderstanding :)

**Review by ggwpez — APPROVED** [review]

**Comment by ggwpez** [comment]
I think this should get a prdoc, since it does change the default behaviour of many crates for downstream users.

File: prdoc/pr_3654.prdoc [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Schema: Polkadot SDK PRDoc Schema (prdoc) v1.0.0
+# See doc at https://raw.githubusercontent.com/paritytech/polkadot-sdk/master/prdoc/schema_user.json
+
+title: Remove `experimental` feature from `pallet-aura`
+
+doc:
+  - audience: Node Dev
```
Comment (bkchr) [inline prdoc/pr_3654.prdoc:7]:
```suggestion
  - audience: Runtime Dev
```

**Review by bkchr — APPROVED** [review]
Thank you!

**Pull request merged**